frontend/packages/noobaa-storage-plugin/src/components/buckets-card/buckets-card.tsx

```diff
@@ -131,48 +115,75 @@ const ObjectDashboardBucketsCard: React.FC<DashboardItemProps> = ({
   const obcLoaded = _.get(resources.obc, 'loaded');
   const obcLoadError = _.get(resources.obc, 'loadError');
 
-  const noobaaStorageClassList = getNoobaaStorageClassList(storageClassesData);
-  const noobaaObcList = getNoobaaObcList(obcData, noobaaStorageClassList);
+  const obData = _.get(resources.ob, 'data', []) as K8sResourceKind[];
+  const obLoaded = _.get(resources.ob, 'loaded');
+  const obLoadError = _.get(resources.ob, 'loadError');
 
-  const obcProvisioningFailure = noobaaObcList.filter((obc) => !isBound(obc)).length;
-
-  const bucketProps: BucketsItemProps = {
-    title: 'Noobaa Bucket',
-    bucketsCount: +getGaugeValue(obCountResponse),
-    objectsCount: getGaugeValue(obObjectsCountResponse),
-    unhealthyCounts: [+getGaugeValue(unhealthyObResponse)],
-    isLoading: !(obCountResponse && obObjectsCountResponse && unhealthyObResponse),
-    hasLoadError: obCountResponseError || obObjectsCountResponseError || unhealthyObResponseError,
-    links: [noobaaBucketsLink],
-  };
+  const noobaaStorageClassList = getNoobaaStorageClassList(storageClassesData);
+  const obcObjects = getGaugeValue(obcObjectsCount);
+  const noobaaObjects = getGaugeValue(noobaaObjectsCount);
 
-  const bucketClaimProps: BucketsItemProps = {
-    title: 'Object Bucket Claim',
-    bucketsCount: noobaaObcList.length,
-    objectsCount: getGaugeValue(obcObjectsCountsResponse),
-    unhealthyCounts: [+getGaugeValue(unhealthyObcCountResponse), obcProvisioningFailure],
-    isLoading: !(
-      storageClassesLoaded &&
-      obcLoaded &&
-      obcObjectsCountsResponse &&
-      unhealthyObcCountResponse
-    ),
-    hasLoadError:
-      storageClassesLoadError ||
-      obcLoadError ||
-      obcObjectsCountsResponseError ||
-      unhealthyObcCountResponseError,
-    links: [noobaaBucketsLink, resourcePathFromModel(NooBaaObjectBucketClaimModel)],
-  };
+  const obcLink = resourcePathFromModel(NooBaaObjectBucketClaimModel);
+  const obLink = resourcePathFromModel(NooBaaObjectBucketModel);
 
   return (
     <DashboardCard>
       <DashboardCardHeader>
         <DashboardCardTitle>Buckets</DashboardCardTitle>
       </DashboardCardHeader>
       <DashboardCardBody>
-        <BucketsItem {...bucketProps} />
-        <BucketsItem {...bucketClaimProps} />
+        <InventoryItem
+          isLoading={!(storageClassesLoaded && noobaaCount)}
+          error={!!(storageClassesLoadError || noobaaCountError)}
+          title="Noobaa Bucket"
+          count={+getGaugeValue(noobaaCount)}
```

Comment:
```suggestion
          count={Number(getGaugeValue(noobaaCount))}
```
We should not rely on JS type coercion for conversion.

Reply:
I guess `Number` is also JS type coercion ?

Reply:
No. That is type casting.

Reply:
Ah, my bad here, you meant to say to use explicit conversion over implicit.Gotcha!
A followup question (just for curiosity), why we should not rely on js type coercion ?
Is the concern wrt precision issue ? 

Thanks!

Reply:
Wouldn't it be better to do the conversion in https://github.com/openshift/console/blob/master/frontend/packages/noobaa-storage-plugin/src/utils.ts#L9 ? You would avoid doing on multiple places

Reply:
Thanks for suggestion, it was like that initially in 4.2 .
Then we had to check the value of guage if its `NaN` or `undefined` at some places hence we removed that. Here its a special case, and the effect of conversion does not matter as inventory card itself handles all !